### PR TITLE
Implement checkstyle and add stubs for docs that were missing

### PR DESCRIPTION
**Pull request by Aphoh**
3 things
1. Now ./gradlew check and ./gradlew checkstyleMain will go through the formatting of every file in src to make sure it is up to the standard laid out in config/checkstyle.xml. Currently google geometry files and generated files are ignored.
2. Docs are corrected for some classes but overall they need a checkup. I don't know what each class  does so some of the added in docs are generated and may need a spicing up.
3. PRs will now fail if they don't meet the checkstyle standards, as ./gradlew check fails on any violations

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -42,11 +45,11 @@ processResources {
 // Run this task to bundle all needed dependency
 task bundle(type: Jar) {
     baseName = project.name + '_bundle'
-    
+
     from {
-    	configurations.compile.collect {
-    		it.isDirectory() && !it.isEmpty() ? it : zipTree(it)
-    	}
+        configurations.compile.collect {
```
Comment (jabbink) [inline build.gradle:50]: Indentation. Ironic, for this PR :joy:

File: build.gradle [hunk 2]
```diff
@@ -59,6 +62,37 @@ protobuf {
     }
 }
 
+def checkstyleOutputDir = "${project.rootDir}/build/reports/checkstyle/"
+
+checkstyle {
+    toolVersion = '7.0'
```
Comment (jabbink) [inline build.gradle:68]: more indentation

File: config/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,207 @@
+<?xml version="1.0"?>
```
Comment (jabbink) [inline config/checkstyle.xml:1]: indentation

**Comment by Aphoh** [comment]
@jabbink when you say indentation you mean what?

File: src/main/java/com/pokegoapi/api/PokemonGo.java [hunk 4]
```diff
@@ -73,6 +77,12 @@
 
 	private long lastInventoryUpdate;
 
+	/**
+	 * Instantiates a new Pokemon go.
+	 *
+	 * @param auth   the auth
+	 * @param client the client
+	 */
```
Comment (gwelican) [inline src/main/java/com/pokegoapi/api/PokemonGo.java:85]: Altough for public API it is good to have documentation, but do you think this gives anything more than the actual code?

looking at the constructor(given that everyone uses IDE nowadays), you can see it's a pokemongo class, takes authinfo and okhttpclient

File: src/main/java/com/pokegoapi/main/RequestHandler.java [hunk 5]
```diff
@@ -130,23 +150,30 @@ public void sendServerRequests() throws RemoteServerException, LoginFailedExcept
 	private void resetBuilder() {
 		builder = RequestEnvelopeOuterClass.RequestEnvelope.newBuilder();
 		builder.setStatusCode(2);
-		builder.setRequestId(8145806132888207460l);
-		if (lastAuth != null && lastAuth.getExpireTimestampMs() > 0)
+		builder.setRequestId(8145806132888207460L);
```
Comment (gwelican) [inline src/main/java/com/pokegoapi/main/RequestHandler.java:153]: :heart:

**Comment by gwelican** [comment]
I like the checkstyle part

**Comment by Aphoh** [comment]
@etheld the idea isnt to implement the documentation with this pr, it's to implement a system which will encourage documentation and proper formatting in the future

**Pull request merged**